tracer/src/Datadog.Trace/TracerManager.cs

```diff
@@ -437,5 +443,29 @@ private static void HeartbeatCallback(object state)
             // send traces to the Agent
             _instance?.Statsd?.Gauge(TracerMetricNames.Health.Heartbeat, Tracer.LiveTracerCount);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void WriteTrace(ArraySegment<Span> trace)
+        {
+            foreach (var processor in TraceProcessors)
+            {
+                if (processor is not null)
+                {
+                    try
+                    {
+                        trace = processor.Process(trace);
+                    }
+                    catch (Exception e)
+                    {
+                        Log.Error(e, e.Message);
+                    }
+                }
+            }
+
+            if (trace.Count > 0)
```

Comment:
I know this is an unusual case, but would it make sense to wrap all of the logic inside the `trace.Count > 0` check? If there are no spans at all, I'm not sure what there is to run through the processors

Reply:
On second thought, perhaps a processor could theoretically add a span to the ArraySegment if it is empty, so nevermind the current behavior seems fine. I'll leave this comment unresolved if anyone has additional insights

Reply:
Yes that was the intention, a processor could generate new spans in theory, but I don’t have a strong use case. So I don’t have a strong opinion here.